### PR TITLE
Add how to add filters to tab menu example see issue

### LINKED ISSUE #2607: Docs improvement how to add filters to tab menu
Since the tab menu is part of the list action it should be mentioned in the docs that you can use the `configureTabMenu` method for adding filters. For instance:

```php
use Sonata\AdminBundle\Admin\Admin;
use Knp\Menu\ItemInterface as MenuItemInterface;
use Sonata\CoreBundle\Form\Type\EqualType;

class FooAdmin extends Admin
{
    protected function configureTabMenu(MenuItemInterface $menu, $action, AdminInterface $childAdmin = null)
    {
        if (!$childAdmin && !in_array($action, array('edit', 'show', 'list'))) {
            return;
        }

        if ($action == 'list') {
            $filterParameters = $this->getFilterParameters();

            // Add or override filter parameters
            $filterParameters['bar'] = array(
                        'type'  => EqualType::TYPE_IS_EQUAL, // equal to
                        'value' => 'baz'
                    );

            // Add filters to uri of tab
            $menu->addChild('Show Baz only', array('uri' => $this->generateUrl('list', array(
                    'filter' => $filterParameters
                ))));

            return;
        }
    }
}
```

Should I create a PR for this section?
http://sonata-project.org/bundles/admin/master/doc/reference/advanced.html#dropdowns-in-tab-menu

It could also be added to the custom `CRUDController` chapters:
- http://sonata-project.org/bundles/doctrine-orm-admin/master/doc/tutorial/creating_your_first_admin_class/defining_crud_controller.html. 
- http://sonata-project.org/bundles/admin/master/doc/cookbook/recipe_custom_action.html

Example usage:
```php
return new RedirectResponse($this->admin->generateUrl('list', array('filter' => $this->admin->getFilterParameters())));
```
Source: http://russische-musik.vs137150.vserver.de/v/Sonata/AdminBundle/Controller/CRUDController:addFlash

Related to:
- https://github.com/sonata-project/SonataAdminBundle/issues/2529
- https://github.com/sonata-project/SonataAdminBundle/issues/1556
- https://github.com/sonata-project/SonataAdminBundle/issues/1556

### PR DESCRIPTION
**Pull request by OskarStark**
Fixes #2607
Fixes #2807 

I applied your comments from #2807 @Soullivaneuh and moved the doc block to the correct section.
I kept the commit of @webdevilopers 

@greg0ire @core23 @Soullivaneuh , can someone please merge this?

### CODE REVIEW

File: Resources/doc/reference/advanced_configuration.rst [hunk 1]
```diff
@@ -225,6 +225,51 @@ You will just need to change the way forms are configured in order to take into
             $formMapper->add('year', 'integer');
         }
     }
+    
+
+Filter parameters in Tab Menu
+---------------------
```
Comment (greg0ire) [inline Resources/doc/reference/advanced_configuration.rst]: "Menu" is going to fall down

File: Resources/doc/reference/advanced_configuration.rst [hunk 2]
```diff
@@ -225,6 +225,51 @@ You will just need to change the way forms are configured in order to take into
...
+
+You can add or override filter parameters to the Tab Menu:
+
+.. code-block:: php
+
+    <?php
+    use Sonata\AdminBundle\Admin\Admin;
+    use Sonata\AdminBundle\Admin\AdminInterface;
+    use Knp\Menu\ItemInterface as MenuItemInterface;
+    use Sonata\CoreBundle\Form\Type\EqualType;
```
Comment (greg0ire) [inline Resources/doc/reference/advanced_configuration.rst]: Please sort this #OCD

**Comment by OskarStark** [comment]
fixed cs

File: Resources/doc/reference/advanced_configuration.rst [hunk 3]
```diff
@@ -306,6 +306,51 @@ inherit it :
...
+                $filterParameters = $this->getFilterParameters();
+
+                // Add or override filter parameters
+                $filterParameters['status'] = array(
+                    'type'  => EqualType::TYPE_IS_EQUAL, // => 1
+                    'value' => Delivery::STATUS_OPEN,
+                );
+
+                // Add filters to uri of tab
+                $menu->addChild('List open deliveries', array(^'uri' => $this->generateUrl('list', array(
```
Comment (core23) [inline Resources/doc/reference/advanced_configuration.rst]: Sure the ``^`` is needed?

Comment (Soullivaneuh) [inline Resources/doc/reference/advanced_configuration.rst]: I don't think so. :-)

**Comment by OskarStark** [comment]
fixxed @greg0ire

File: Resources/doc/reference/advanced_configuration.rst [hunk 4]
```diff
@@ -306,6 +306,51 @@ inherit it :
...
+
+You can add or override filter parameters to the Tab Menu:
+
+.. code-block:: php
+
+    <?php
+
+    use Sonata\AdminBundle\Admin\Admin;
+    use Sonata\AdminBundle\Admin\AdminInterface;
+    use Knp\Menu\ItemInterface as MenuItemInterface;
```
Comment (greg0ire) [inline Resources/doc/reference/advanced_configuration.rst]: Sorry, but this is still not sorted.

Comment (OskarStark) [inline Resources/doc/reference/advanced_configuration.rst]: fixxed

**Pull request merged**